### PR TITLE
Project

### PR DESCRIPTION
**Pull request by vvale93**
Work in progress

### CODE REVIEW

File: Proyecto/BootcampProyect/src/app/char-detail/char-detail.component.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    private RaceMatcherService: RaceMatcherService,
+    private ClassMatcherService: ClassMatcherService,
+    private PvpInfoUserService: PvpInfoUserService
+  ) { }
+
+  ngOnInit() {
+    let name: string;
+    let realm: string;
+    this.route.params.subscribe((params) => {
+      realm = params['realm'];
```
Comment (DemianMoschin-GLB) [inline Proyecto/BootcampProyect/src/app/char-detail/char-detail.component.ts]: Are you trying to get an object value by array annotation? Don't use array annotation for object getters

File: Proyecto/BootcampProyect/src/app/char-detail/char-detail.component.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  }
+  private getCharacter(name:string, realm:string) {
+    this.service.getChar(name, realm).subscribe((char) => {
+      this.char = char;
+      this.imagePath = `http://render-us.worldofwarcraft.com/character/${char.thumbnail}`;   
+      this.raceId = char.race;  
+      this.classId = char.class;
+      this.RaceMatcherService.getRaces().subscribe(races => {
+        this.races = races;
+        this.race = races.find(x => x.id == this.raceId);
```
Comment (DemianMoschin-GLB) [inline Proyecto/BootcampProyect/src/app/char-detail/char-detail.component.ts]: equal operator is not right
(always use ===)

File: Proyecto/BootcampProyect/src/app/class-matcher.service.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import { Http } from '@angular/http';
+import { Class } from './class';
+import 'rxjs/add/operator/map';
+
+@Injectable()
+export class ClassMatcherService {
+
+  constructor(private http: Http) { }
+  getClasses() : Observable<Class[]> {
+    return this.http.get(`https://us.api.battle.net/wow/data/character/classes?locale=en_US&apikey=68rhxtstmd6ksae9jtxeywtexvm2atsd`)
```
Comment (DemianMoschin-GLB) [inline Proyecto/BootcampProyect/src/app/class-matcher.service.ts]: Extract the key in another file, so you can share it across your app

File: Proyecto/BootcampProyect/src/app/guild.service.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import { Guild } from './guild';
+import 'rxjs/add/operator/map';
+
+@Injectable()
+export class GuildService {
+
+  constructor(private http : Http) { }
+  getGuild(guildName:string, guildRealm:string) : Observable<Guild> {
+    guildName =  encodeURI(guildName);
+    return this.http.get(`https://us.api.battle.net/wow/guild/${guildRealm}/${guildName}?locale=en_US&apikey=68rhxtstmd6ksae9jtxeywtexvm2atsd`)
```
Comment (DemianMoschin-GLB) [inline Proyecto/BootcampProyect/src/app/guild.service.ts]: same as above

File: Proyecto/BootcampProyect/src/app/pvp-info-user.service.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import { UserItems } from './user-items'
+import 'rxjs/add/operator/map';
+
+@Injectable()
+export class PvpInfoUserService {
+
+  constructor(private http : Http) { }
+
+  getInfoPvp(charName: string, charRealm: string) : Observable<PvpInfoUser> {
+    return this.http.get(`https://us.api.battle.net/wow/character/${charRealm}/${charName}?fields=pvp&locale=en_US&apikey=68rhxtstmd6ksae9jtxeywtexvm2atsd`)
```
Comment (DemianMoschin-GLB) [inline Proyecto/BootcampProyect/src/app/pvp-info-user.service.ts]: Remember to reuse inside constants pieces of this string you're using across this functions,. same as comments above

File: Proyecto/BootcampProyect/src/app/searchs/searchs.component.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  char : Char;
+  guild: Guild;
+  constructor(private SearchCharService: SearchCharService, private GuildService: GuildService) { }
+
+  ngOnInit() {
+    
+  }
+  getCharacter(charName: string, charRealm: string) {
+    if(!charName || !charRealm) {
```
Comment (DemianMoschin-GLB) [inline Proyecto/BootcampProyect/src/app/searchs/searchs.component.ts]: If you're expecting both not to be null, use &&

**Comment by DemianMoschin-GLB** [comment]
Great work Vale!

Just a couple of things: 
Inside functions, you can declare variables at the beginning of your function body.  Don't mix declarations and assignations, unless you need it.

File: Proyecto/BootcampProyect/src/app/char-detail/char-detail.component.ts (hunk 1 above)
Comment (vvale93) [reply Proyecto/BootcampProyect/src/app/char-detail/char-detail.component.ts]: done

File: Proyecto/BootcampProyect/src/app/class-matcher.service.ts (hunk 3 above)
Comment (vvale93) [reply Proyecto/BootcampProyect/src/app/class-matcher.service.ts]: done

File: Proyecto/BootcampProyect/src/app/guild.service.ts (hunk 4 above)
Comment (vvale93) [reply Proyecto/BootcampProyect/src/app/guild.service.ts]: done

File: Proyecto/BootcampProyect/src/app/pvp-info-user.service.ts (hunk 5 above)
Comment (vvale93) [reply Proyecto/BootcampProyect/src/app/pvp-info-user.service.ts]: done

File: Proyecto/BootcampProyect/src/app/searchs/searchs.component.ts (hunk 6 above)
Comment (vvale93) [reply Proyecto/BootcampProyect/src/app/searchs/searchs.component.ts]: done

Comment (vvale93) [reply Proyecto/BootcampProyect/src/app/searchs/searchs.component.ts]: I'm expecting to emit the alert when one or both fields are null

File: Proyecto/BootcampProyect/src/app/char-detail/char-detail.component.ts (hunk 2 above)
Comment (vvale93) [reply Proyecto/BootcampProyect/src/app/char-detail/char-detail.component.ts]: done

**Pull request merged**